operators/src/processing/point_in_polygon.rs

```diff
@@ -0,0 +1,775 @@
+use futures::stream::BoxStream;
+use futures::{StreamExt, TryStreamExt};
+use snafu::ensure;
+
+use geoengine_datatypes::collections::{
+    MultiPointCollection, MultiPolygonCollection, VectorDataType,
+};
+use geoengine_datatypes::primitives::{Coordinate2D, TimeInterval};
+
+use crate::adapters::FeatureCollectionChunkMerger;
+use crate::engine::{
+    ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedVectorOperator,
+    Operator, QueryContext, QueryProcessor, QueryRectangle, TypedVectorQueryProcessor,
+    VectorOperator, VectorQueryProcessor, VectorResultDescriptor,
+};
+use crate::error;
+use crate::util::Result;
+use arrow::array::BooleanArray;
+use itertools::Itertools;
+
+pub type PointInPolygonFilter = Operator<()>;
+
+#[typetag::serde]
+impl VectorOperator for PointInPolygonFilter {
+    fn initialize(
+        self: Box<Self>,
+        context: &ExecutionContext,
+    ) -> Result<Box<InitializedVectorOperator>> {
+        ensure!(
+            self.vector_sources.len() == 2,
+            error::InvalidNumberOfVectorInputs {
+                expected: 2..3,
+                found: self.vector_sources.len()
+            }
+        );
+        ensure!(
+            self.raster_sources.is_empty(),
+            error::InvalidNumberOfRasterInputs {
+                expected: 0..1,
+                found: self.raster_sources.len()
+            }
+        );
+
+        let vector_sources = self
+            .vector_sources
+            .into_iter()
+            .map(|o| o.initialize(context))
+            .collect::<Result<Vec<Box<InitializedVectorOperator>>>>()?;
+
+        ensure!(
+            vector_sources[0].result_descriptor().data_type == VectorDataType::MultiPoint,
```

Comment:
why is the order of the inputs important?

Reply:
It's Point-in-Polygon, the first operator is points and the second polygons.

Reply:
but how can the user know which order to input? we could also just require that one input is points and one input ins polygons and extend the code to be able to handle that.

Reply:
I've created the comment: 
362d4e4 

But for the thing about named input, this PR is not the place to change our default implementation of `Operator`.